onnx/defs/schema.h

```diff
@@ -205,6 +214,13 @@ class OpSchema {
   OpSchema& EnforceConsumed(std::unordered_map<int, int> inplace);
   OpSchema& EnforceOneToOneConsumed();
 
+  // Shape Inference
+  //
+  // Note that signatures are defined to allow for forward-declaring
+  // any structs used from ir.h
+  OpSchema& ShapeInferenceFunction(InferenceFunction inferenceFunction);
+  std::vector<TypeProto_Tensor> ApplyShapeInferenceFunction(InferenceContext& inferenceContext) const;
```

Comment:
minor suggestion: Just return the function and let the caller to call the function?

Reply:
yes, that will be better